cmd/thanos/store.go

```diff
@@ -381,14 +386,25 @@ func runStore(
 
 			level.Info(logger).Log("msg", "initializing bucket store")
 			begin := time.Now()
-			if err := bs.InitialSync(ctx); err != nil {
+
+			//This will stop retrying after 15 seconds.
```

Comment:
The value `15 seconds` can change in future which will make the comment factually incorrect. Please use:
 .... after set timeout duration.